x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/lib/create_rate_agg_with_interface.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { InfraTimerangeInput } from '../../../../../common/http_api';
+import { calculateRateTimeranges } from './calculate_rate_timeranges';
+
+export const createRateAggsWithInterface = (
+  timerange: InfraTimerangeInput,
+  id: string,
+  field: string,
+  interfaceField: string
+) => {
+  const { firstBucketRange, secondBucketRange, intervalInSeconds } =
+    calculateRateTimeranges(timerange);
+
+  const interfaceAggs = {
+    interfaces: {
+      terms: {
+        field: interfaceField,
+      },
+      aggs: { maxValue: { max: { field } } },
+    },
+    sumOfInterfaces: {
+      sum_bucket: {
+        buckets_path: 'interfaces>maxValue',
```

Comment:
@simianhacker If I understand correctly, here you calculate the sum of the maxValue across all your interface buckets? I am trying to understand what `the sum of max values` can bring?

I can understand for example the sum of all the total monthly sales buckets for example https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-sum-bucket-aggregation.html, but not the sum of max values of all the total interfaces buckets.

Reply:
The sum of the max of all the interfaces will give us a single number that represents all the bytes transfer across all the interfaces at that point in time (remember it's a monatomic increasing number). Then when we take the difference of these two buckets, that will give us the total bytes transferred during the second bucket. Finally we need to reduce the rate down to a "per second" rate by dividing the difference by the total number of seconds 1 bucket represents. 

Here's a blog post I wrote on the different types of rates and visualizing them with Kibana: https://www.elastic.co/es/blog/visualizing-observability-with-kibana-event-rates-and-rate-of-change-in-tsvb

Reply:
@simianhacker Thanks a ton! This `remember it's a monatomic increasing number` answers exactly my question!

Reply:
Checking out your blog post now! Respect!